File: csharp/2016/the-magnificent-seven-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "The Magnificent Seven" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "The Magnificent Seven",
                year: 2016,
                plot: "Seven gunmen in the old west gradually come together to help a poor village against savage thieves.",
                rating: 6.9
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "The Magnificent Seven",
                year: 2016
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2016/the-magnificent-seven-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add The Magnificent Seven to the database
movies.insert(
    title="The Magnificent Seven", 
    year=2016, 
    plot="Seven gunmen in the old west gradually come together to help a poor village against savage thieves.", 
    rating=6.9
)

# Confirm that the movie was added
movie = movies.select(
    title="The Magnificent Seven", 
    year=2016
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    